File: backend/controllers/aktivnostiController.js
const aktivnostiService = require('../services/aktivnostiService');

class AktivnostiController {
  async getAll(req, res) {
    try {
      const aktivnosti = await aktivnostiService.getAllAktivnosti();
      res.json(aktivnosti);
    } catch (error) {
      res.status(500).json({ message: error.message });
    }
  }

  async getById(req, res) {
    try {
      const aktivnost = await aktivnostiService.getAktivnostById(parseInt(req.params.id));
      res.json(aktivnost);
    } catch (error) {
      res.status(404).json({ message: error.message });
    }
  }

  async create(req, res) {
    try {
      const aktivnost = await aktivnostiService.createAktivnost(req.body);
      res.status(201).json(aktivnost);
    } catch (error) {
      res.status(400).json({ message: error.message });
    }
  }

  async update(req, res) {
    try {
      const aktivnost = await aktivnostiService.updateAktivnost(parseInt(req.params.id), req.body);
      res.json(aktivnost);
    } catch (error) {
      res.status(400).json({ message: error.message });
    }
  }

  async delete(req, res) {
    try {
      const aktivnost = await aktivnostiService.deleteAktivnost(parseInt(req.params.id));
      res.json(aktivnost);
    } catch (error) {
      res.status(404).json({ message: error.message });
    }
  }
}

module.exports = new AktivnostiController();

File: JavaSimpleBackend/src/main/java/com/atesti/controller/AktivnostiController.java
package com.atesti.controller;

import com.atesti.dto.AktivnostRequest;
import com.atesti.entity.Aktivnost;
import com.atesti.service.AktivnostiService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class AktivnostiController {

    private final AktivnostiService aktivnostiService;

    @GetMapping({"/api/aktivnosti", "/aktivnosti"})
    public ResponseEntity<List<Aktivnost>> getAll() {
        return ResponseEntity.ok(aktivnostiService.getAllAktivnosti());
    }

    @GetMapping({"/api/aktivnosti/{id}", "/aktivnosti/{id}"})
    public ResponseEntity<Aktivnost> getById(@PathVariable Long id) {
        return ResponseEntity.ok(aktivnostiService.getAktivnostById(id));
    }

    @PostMapping({"/api/aktivnosti", "/aktivnosti"})
    public ResponseEntity<Aktivnost> create(@RequestBody AktivnostRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(aktivnostiService.createAktivnost(request));
    }

    @PutMapping({"/api/aktivnosti/{id}", "/aktivnosti/{id}"})
    public ResponseEntity<Aktivnost> update(@PathVariable Long id, @RequestBody AktivnostRequest request) {
        return ResponseEntity.ok(aktivnostiService.updateAktivnost(id, request));
    }

    @DeleteMapping({"/api/aktivnosti/{id}", "/aktivnosti/{id}"})
    public ResponseEntity<Aktivnost> delete(@PathVariable Long id) {
        return ResponseEntity.ok(aktivnostiService.deleteAktivnost(id));
    }
}
